bin/utils.mjs

```diff
@@ -1,9 +1,23 @@
 import createMarkdownLoader from '../src/loaders/markdown.mjs';
 import createMarkdownParser from '../src/parsers/markdown.mjs';
 
-// Instantiate loader and parser once to reuse
-const loader = createMarkdownLoader();
-const parser = createMarkdownParser();
+/**
+ * Generic lazy initializer.
+ * @template T
+ * @param {() => T} factory - Function to create the instance.
+ * @returns {() => T} - A function that returns the singleton instance.
+ */
+export const lazy = factory => {
+  let instance;
+  return () => (instance ??= factory());
+};
+
+// Instantiate loader and parser once to reuse,
+// but only if/when we actually need them. No need
+// to create these objects just to load a different
+// utility.
+const loader = lazy(createMarkdownLoader);
+const parser = lazy(createMarkdownParser);
```

Comment:
These have changed because there's no need to run these functions when the user isn't actually going to use them (now that the `utils` file has more than just the parsing utility)